includes/class-sensei-learner.php

```diff
@@ -9,13 +9,68 @@
  * @since 1.9.0
  */
 class Sensei_Learner {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
 	/**
 	 * Cache of the learner terms.
 	 *
 	 * @var WP_Term[]
 	 */
 	private static $learner_terms = [];
 
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Course_Enrolment_Manager constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Sets the actions.
+	 *
+	 * @since 3.0.0
+	 */
+	public function init() {
+		// Delete user activity and enrolment terms when user is deleted.
+		add_action( 'deleted_user', array( $this, 'delete_user_registers' ) );
+	}
+
+	/**
+	 * Delete all user activity and enrolment terms when user is deleted.
+	 *
+	 * @param  integer $user_id User ID.
+	 * @return void
+	 */
+	public function delete_user_registers( $user_id = 0 ) {
+		if ( $user_id ) {
+			// Remove activities.
+			Sensei_Utils::delete_all_user_activity( $user_id );
```

Comment:
I also thought about to move this method to this class, because it's used only here. But I gave up because it's a public static method and we'd have to deprecate it on `Sensei_Utils` and duplicate here.

And if we need it again in another place we would have to undo it.

Reply:
I like the idea of moving it to this class. We could retain the method as public and call it from the deprecated method like this: `Senei_Lerner::instance()->delete_user_registers`. We could also warn the users to use `delete_user_registers` instead. If we leave it like this then users calling the `delete_all_user_activity` will have leftover data in their db.

Reply:
Good catch!

https://github.com/Automattic/sensei/pull/2898/commits/fcec99649d51c0b1d1e3c4689733829be1aa8892

Reply:
I liked your approach! I did it here: https://github.com/Automattic/sensei/pull/2898/commits/26057ed2bc52cf99db56b951106fbc0727cd2c23